hal/src/nRF52840/interrupts_hal.cpp

```diff
@@ -141,25 +142,26 @@ void HAL_Interrupts_Attach(uint16_t pin, HAL_InterruptHandler handler, void* dat
         }
     } else if (err_code == NRFX_ERROR_NO_MEM) {
         // All channels have been used up
-        return;
+        return SYSTEM_ERROR_NO_MEMORY;
     }
 
     nrfx_gpiote_in_event_enable(nrf_pin, true);
+    return SYSTEM_ERROR_NONE;
 }
 
-void HAL_Interrupts_Detach(uint16_t pin) {
-    HAL_Interrupts_Detach_Ext(pin, 0, NULL);
+int HAL_Interrupts_Detach(uint16_t pin) {
+    return HAL_Interrupts_Detach_Ext(pin, 0, NULL);
 }
 
-void HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved) {
+int HAL_Interrupts_Detach_Ext(uint16_t pin, uint8_t keepHandler, void* reserved) {
     if (keepHandler) {
```

Comment:
This is actually a bug, and `keepHandler` isn't correctly handled. I'll commit to this branch to fix this.

Reply:
Fixed in https://github.com/particle-iot/device-os/pull/1939/commits/1e6ad02d278d05dfecfe86180a1da71f9cb5919e

Reply:
Great, thanks!